README.md

```diff
@@ -0,0 +1,209 @@
+- **CategoryService**: Manages category processing logic, including category management for admins and category retrieval for users.
+- **ShoppingCartService**: Handles the business logic related to shopping carts, including adding items and updating the cart.
+- **UserService**: Contains business logic for user operations such as registration and user retrieval by username (email).
+
+### Repositories
+
+- **BookRepository**: Interface for CRUD operations on Book entities and special methods for finding books by category.
+- **ShoppingCartRepository**: Interface for CRUD operations on ShoppingCart entities and methods for identifying a shopping cart by user ID.
+- **CartItemRepository**: Interface for CRUD operations on CartItem entities.
+- **OrderRepository**: Interface for CRUD operations on Order entities.
+- **OrderItemRepository**: Interface for CRUD operations on OrderItem entities and methods for retrieving order items by order ID.
+- **UserRepository**: Interface for CRUD operations on User entities and methods for checking if the user is already registered.
+- **RoleRepository**: Interface for CRUD operations on Role entities, including methods for identifying user roles.
+
+## Setup and Usage
+
+To get started with this project, follow these steps:
+
+1. **Clone the repository**:
+    ```sh
+    git clone https://github.com/bodya4243/online-book-store.git
+    cd online-book-store
+    ```
+
+2. **Build the project using Maven**:
+    ```sh
+    mvn clean install
+    ```
+
+3. **Run the application**:
+    ```sh
+    mvn spring-boot:run
+    ```
+   Technologies and Tools
+   Key Technologies
+   Java Development Kit (JDK) 17
+   Maven
+   MySQL
+   Spring Boot
+   H2 Database (for testing)
+   Lombok
+   MapStruct
+   Liquibase
+   SpringDoc OpenAPI
+   JWT (JSON Web Token)
+   Docker Compose
+   
+
+
+4. **Access the application**:
+   - The application runs on: http://localhost:8080.
+   - API documentation is available at: http://localhost:8080/swagger-ui.html.
+
+### Docker Setup
+
+**Swagger**:
+   - You can use Swagger for manipulating API endpoints.
+
+**Docker**:
+   - You can download the container from Docker and start the API.
+
+# MySQL environment variables
+```
+   DB_PASSWORD=123sqlbod321
+   DB_DATABASE=book_store_db
+   DB_USER=bodya
+   DB_LOCAL_PORT=3309
+   DB_DOCKER_PORT=3306
+
+   SPRING_LOCAL_PORT=8080
+   SPRING_DOCKER_PORT=8080
+   DEBUG_PORT=5005
+
+   JWT_SECRET=wefwqfdfwqfq4f3fq3af43453452ef34fewfewfe
+   
+   MYSQL_DATABASE=book_store_db
+   MYSQL_USER=bodya
+   MYSQL_PASSWORD=123sqlbod321
+```
+
+1. **Build the Docker image**:
+    ```sh
+    docker build -t online-book-store .
+    ```
+
+2. **Run the Docker container**:
+    ```sh
+    docker run -d -p 8080:8080 online-book-store
+    ```
+
+3. **Verify the container is running**:
+    - Ensure the application is accessible at `http://localhost:8080`.
+
+4. **Authentication and Usage Guide**
```

Comment:
this should not be the 4th point of  "Running the app with a docker" guide, more of a general usage guide which should be separate topic